dpe/src/commands/derive_child.rs

```diff
@@ -64,27 +64,89 @@ impl DeriveChildCmd {
         self.flags.contains(DeriveChildFlags::INPUT_ALLOW_X509)
     }
 
-    /// Check if this will result in two default contexts in the same locality.
+    /// Whether it is okay to make a default context.
     ///
-    /// There can only be one default context in each locality at any given time. A default context
-    /// has a specific handle. If there are multiple, the DPE instance will not be able to
-    /// differentiate between them.
+    /// When a default context is in a locality, it MUST be the only context in the locality. This
+    /// checks that the operation will result in the locality will only have the newly created
+    /// default context and no others.
     ///
     /// # Arguments
     ///
     /// * `parent_idx` - Index of the soon-to-be parent.
     /// * `default_context_idx` - Index of the current default context, if there is one.
+    /// * `num_contexts_in_locality` - Number of contexts already in the locality.
     fn safe_to_make_default(
         &self,
         parent_idx: usize,
-        default_context_idx: Result<usize, DpeErrorCode>,
+        default_context_idx: Option<usize>,
+        num_contexts_in_locality: usize,
     ) -> bool {
-        if let Ok(default_idx) = default_context_idx {
-            if default_idx != parent_idx || self.retains_parent() {
-                return false;
+        match (num_contexts_in_locality, default_context_idx) {
+            // No other contexts in the locality.
+            (0, None) => true,
+            // There is only one context, but that context is the parent.
+            (1, Some(default_idx)) if default_idx == parent_idx => {
+                // It is okay if the parent is about to be retired.
+                !self.retains_parent()
```

Comment:
I think whether or not this is ok still depends on whether the parent and child contexts are in the same locality, right?

Reply:
Yeah that's true, default_context_idx is the index of the default context in the target locality so it should be covered. Or am I missing something?

Reply:
Done